src/node/util.ts

```diff
@@ -224,3 +224,51 @@ export function extend(...args: any[]): any {
   }
   return c
 }
+
+/**
+ * Taken from vs/base/common/charCode.ts. Copied for now instead of importing so
+ * we don't have to set up a `vs` alias to be able to import with types (since
```

Comment:
What's hard about setting up a vs alias to import? And why is it necessary? Why can't we just import the file?

I thought the reason for duplication was to avoid bringing in a ton of VS Code.

Reply:
Yeah that's the reasoning for `uri.ts` but `charCode.ts` doesn't import anything and has just the one enum so it wouldn't be too bad (although it's 425 lines).

To import directly we'd have to import from `out` so it works in the final build (`from "../../lib/vscode/out/vs/base/common/charCode`). But that causes type errors (`Could not find a declaration file for module`).

So we could do something like `from "vs/base/common/charCode"` but that means `vs` needs to point to `src` during development and `out` in the build. There are aliases for `tsc` but they don't affect the generated code (I think) so we'd need to hijack `require` or add a dependency to do it for us.

I feel like there might be a good alternative solution but I haven't come upon it yet.

Reply:
> To import directly we'd have to import from out so it works in the final build

Why do we have to import from `out`? Why not `../../lib/vscode/src/vs/base/common/charCode`?